exporter/prometheusexporter/prometheus.go

```diff
@@ -42,12 +44,40 @@ func (pe *prometheusExporter) Start(_ context.Context, _ component.Host) error {
 }
 
 func (pe *prometheusExporter) ConsumeMetricsData(ctx context.Context, md consumerdata.MetricsData) error {
+	merged := make(map[string]*metricspb.Metric)
 	for _, metric := range md.Metrics {
+		merge(merged, metric)
+	}
+	for _, metric := range merged {
 		_ = pe.exporter.ExportMetric(ctx, md.Node, md.Resource, metric)
 	}
 	return nil
 }
 
+// The underlying exporter overwrites timeseries when there are conflicting metric signatures.
+// Therefore, we need to merge timeseries that share a metric signature into a single metric before sending.
+func merge(m map[string]*metricspb.Metric, metric *metricspb.Metric) {
+	key := metricSignature(metric)
+	current, ok := m[key]
+	if !ok {
+		m[key] = metric
+		return
+	}
+	current.Timeseries = append(current.Timeseries, metric.Timeseries...)
+}
+
+// Unique identifier of a given promtheus metric
+// Assumes label keys are always in the same order
+func metricSignature(metric *metricspb.Metric) string {
+	var buf bytes.Buffer
+	buf.WriteString(metric.GetMetricDescriptor().GetName())
+	labelKeys := metric.GetMetricDescriptor().GetLabelKeys()
+	for _, labelKey := range labelKeys {
+		buf.WriteString("-" + labelKey.Key)
```

Comment:
Are dashes allowed in metric names and label keys? If yes then this can result in the same metric signature for different metrics. The following 2 different metrics produce the same signature:
1. Name: abc
    Label key: d-ef
    Signature: abc-d-ef
2. Name: abc-d
    Label key: ef
    Signature: abc-d-ef

Reply:
It is, however, this signature function is a copy-paste from the exporter. I guess it doesn't hurt to change it though, would you like me to do so?

Reply:
This [doc](https://prometheus.io/docs/concepts/data_model/#metric-names-and-labels) seems to indicate dashes aren't allowed in metric names or labels in prometheus.

Reply:
@tigrannajaryan I think we're good to merge then 🙂